packages/dx-scheduler-core/src/plugins/date-navigator/helpers.ts

```diff
@@ -2,43 +2,75 @@ import moment from 'moment';
 import { PureComputed } from '@devexpress/dx-core';
 import { ViewBoundTextFn } from '../../types';
 
+const day = { day: 'numeric' };
+const shortMonth = { month: 'short' };
+const dayShortMonth = { day: 'numeric', month: 'short' };
+const dayShortMonthLongYear = { day: 'numeric', month: 'short', year: 'numeric' };
+const dayShortMonthShortYear = { day: 'numeric', month: 'short', year: '2-digit' };
+const dayLongMonthLongYear = { day: 'numeric', month: 'long', year: 'numeric' };
+const shortMonthLongYear = { month: 'short', year: 'numeric' };
+const shortMonthShortYear = { month: 'short', year: '2-digit' };
+const longMonthLongYear = { month: 'long', year: 'numeric' };
+
 const calculateTextByDays: PureComputed<
-  [Date, Date], string
-> = (startViewDate, endViewDate) => {
+  [Date, Date, any], string
+> = (startViewDate, endViewDate, formatDate) => {
   const momentStartViewDate = moment(startViewDate as Date);
```

Comment:
`startViewDate` is of `Date` type according to function type - why is `as Date` used?

Reply:
TThe `PureComputed` makes a readonly Date. Moment.js don't want a readonly arguments.